### PR TITLE
Add ci.yml github action

### PR DESCRIPTION
**Pull request by ivanahuckova**
BigQuery does not have e2e tests, so this only includes `Build, lint and unit tests` test.

### CODE REVIEW

**Comment by ivanahuckova** [comment]
@yesoreyeram @gwdawson if you have a time, would you check this PR that just adds ci.yml github action so it checks lint, typechecks and unit tests - to prevent us from having type errors.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        run: |
+          mv dist ${{ steps.metadata.outputs.plugin-id }}
+          zip ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.plugin-id }} -r
+
+      - name: Archive Build
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.metadata.outputs.plugin-id }}-${{ steps.metadata.outputs.plugin-version }}
+          path: ${{ steps.metadata.outputs.plugin-id }}
+          retention-days: 5
```
Comment (gabor) [inline .github/workflows/ci.yml:108]: are these parts intentional? to me it seems this builds the package, and uploads it to somewhere... do we need it here?

**Review by gabor — COMMENTED** [review]
code looks fine, but i wonder if it does too many things, could you check my comment please? thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ivanahuckova) [reply .github/workflows/ci.yml:108]: We can keep it more simple, but this is what other plugins use. Let me know what do you think.

Comment (gabor) [reply .github/workflows/ci.yml:108]: let's keep it like this then 👍

**Review by gabor — APPROVED** [review]

**Pull request merged**